File: tests/fixtures/csharp/shadowing.cs
namespace Fixtures.Shadowing;

delegate void Callback();

class Shadowing
{
    Callback Make() => null!;
    Callback Target() => null!;

    void Run()
    {
        Callback target = Target();
        target();

        target = Make();
        target();

        Callback first = Make(), second = Make();
        first();
        second();

        {
            Callback nested = Make();
            nested();
        }
        nested();

        receiver.callback = Make();
        receiver();
        values[0] = Make();
        values();
    }
}

File: tests/fixtures/zig/shadowing.zig
const Callback = *const fn () void;

fn make() Callback {}
fn target() void {}

fn shadowing() void {
    var target = target();
    target();

    target = make();
    target();

    {
        const nested = make();
        nested();
    }
    nested();

    object.callback = make();
    object();
    values[0] = make();
    values();
}
